File: FHD/mobile-android/app/src/main/java/com/xiuci/xcagi/mobile/core/model/AiGroupMessageUi.kt
package com.xiuci.xcagi.mobile.core.model

/** AI 群聊消息气泡 UI：讨论 → 分工 → 派单 → 汇报 → 验收（含误验收待复核）。 */
data class AiGroupMessageKindUi(
    val badge: String,
    val bubbleTone: AiGroupBubbleTone,
    val needsReview: Boolean,
)

enum class AiGroupBubbleTone {
    CHAT,
    DISCUSSION,
    ROUTING,
    WORK,
    ACCEPTANCE,
    ACCEPTANCE_REVIEW,
}

object AiGroupMessageUi {
    fun resolve(kind: String?, status: String?, body: String?): AiGroupMessageKindUi {
        val k = kind.orEmpty().trim().lowercase()
        val bodyText = body.orEmpty()
        return when (k) {
            "discussion", "super_discussion" ->
                AiGroupMessageKindUi("讨论", AiGroupBubbleTone.DISCUSSION, false)
            "routing_decision" ->
                AiGroupMessageKindUi("分工", AiGroupBubbleTone.ROUTING, false)
            "work_order" ->
                AiGroupMessageKindUi("派单", AiGroupBubbleTone.WORK, false)
            "work_report", "relay_work_report" ->
                AiGroupMessageKindUi("汇报", AiGroupBubbleTone.WORK, false)
            "work_acceptance" -> {
                val needsReview =
                    status.orEmpty().trim().lowercase() == "needs_review" || bodyText.contains("需要复核")
                if (needsReview) {
                    AiGroupMessageKindUi("待复核", AiGroupBubbleTone.ACCEPTANCE_REVIEW, true)
                } else {
                    AiGroupMessageKindUi("可验收", AiGroupBubbleTone.ACCEPTANCE, false)
                }
            }
            else -> AiGroupMessageKindUi("", AiGroupBubbleTone.CHAT, false)
        }
    }

    fun sendingLabel(dispatchMode: Boolean): String =
        if (dispatchMode) "员工正在执行并汇报…" else "AI 成员正在讨论并回复…"
}

File: FHD/frontend/src/utils/aiGroupMessageUi.ts
/** AI 群聊消息气泡 UI：按 kind/status 区分讨论、分工、派工、汇报与验收（含误验收待复核）。 */

export type AiGroupMessageKindUi = {
  badge: string;
  bubbleClass: string;
  needsReview: boolean;
};

export function resolveAiGroupMessageUi(
  kind?: string,
  status?: string,
  body?: string,
): AiGroupMessageKindUi {
  const k = String(kind || 'chat').trim().toLowerCase();
  const bodyText = String(body || '');

  if (k === 'discussion' || k === 'super_discussion') {
    return { badge: '讨论', bubbleClass: 'is-discussion', needsReview: false };
  }
  if (k === 'routing_decision') {
    return { badge: '分工', bubbleClass: 'is-routing', needsReview: false };
  }
  if (k === 'work_order') {
    return { badge: '派单', bubbleClass: 'is-work', needsReview: false };
  }
  if (k === 'work_report' || k === 'relay_work_report') {
    return { badge: '汇报', bubbleClass: 'is-work', needsReview: false };
  }
  if (k === 'work_acceptance') {
    const needsReview =
      String(status || '').trim().toLowerCase() === 'needs_review' || bodyText.includes('需要复核');
    return {
      badge: needsReview ? '待复核' : '可验收',
      bubbleClass: needsReview ? 'is-acceptance-review' : 'is-acceptance',
      needsReview,
    };
  }
  return { badge: '', bubbleClass: '', needsReview: false };
}

export function groupSendingLabel(dispatchMode: boolean): string {
  return dispatchMode ? '员工正在执行并汇报…' : 'AI 成员正在讨论并回复…';
}
